tq/transfer_queue.go

```diff
@@ -506,6 +522,19 @@ func (q *TransferQueue) enqueueAndCollectRetriesFor(batch batch) (batch, error)
 	next := q.makeBatch()
 	tracerx.Printf("tq: sending batch of size %d", len(batch))
 
+	enqueueRetry := func(t *objectTuple, err error) {
+		count := q.rc.Increment(t.Oid)
+		t.ReadyTime = q.rc.ReadyTime(t.Oid)
+		delay := time.Until(t.ReadyTime).Seconds()
+
+		var errMsg string
+		if err != nil {
+			errMsg = ": " + err.Error()
+		}
+		tracerx.Printf("tq: enqueue retry #%d after %.2fs for %q (size: %d)%s", count, delay, t.Oid, t.Size, errMsg)
+		next = append(next, t)
```

Comment:
I take it this is written inline and not as an independent method on the queue so that we can close over `next`.

Reply:
Yes, but I can also make it an independent method on the queue that takes a `*batch` if you prefer that style.

Reply:
It's fine as it is.  I just wanted to be clear on what's going on.